src/main/java/de/hysky/skyblocker/skyblock/chat/chatcoords/ChatWaypointLocation.java

```diff
@@ -0,0 +1,63 @@
+package de.hysky.skyblocker.skyblock.chat.chatcoords;
+
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.utils.Constants;
+import de.hysky.skyblocker.utils.Utils;
+import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.text.ClickEvent;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ChatWaypointLocation {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ChatWaypointLocation.class);
+    private static final Pattern SKYBLOCKER_COORDS_PATTERN = Pattern.compile("x: (-?[0-9]+), y: ([0-9]+), z: (-?[0-9]+)");
+    private static final Pattern SKYHANNI_DIANA_PATTERN = Pattern.compile("(?<party>§9Party §8> )?(?<playerName>.+)§f: §rA MINOS INQUISITOR has spawned near \\[(?<area>.*)] at Coords (?<x>[^ ]+) (?<y>[^ ]+) (?<z>[^ ]+)");
+
+    private static final List<Pattern> PATTERNS = List.of(SKYBLOCKER_COORDS_PATTERN, SKYHANNI_DIANA_PATTERN);
+
+    public static void init() {
+        ClientReceiveMessageEvents.GAME.register(ChatWaypointLocation::onMessage);
+    }
+
+    private static void onMessage(Text text, boolean overlay) {
+        if (Utils.isOnSkyblock() && SkyblockerConfigManager.get().uiAndVisuals.waypoints.enableWaypoints) {
+
+            String message = text.getString();
+
+            for (Pattern pattern : PATTERNS) {
+                Matcher matcher = pattern.matcher(message);
+                if (matcher.find()) {
+                    try {
+                        if (pattern == SKYBLOCKER_COORDS_PATTERN) {
+                            // Call requestWaypoint for the first pattern
+                            requestWaypoint(matcher.group(1), matcher.group(2), matcher.group(3));
```

Comment:
Just make `SKYBLOCKER_COORDS_PATTERN` have named capturing groups too so you don't need two types of handling code.